### ISSUE TITLE
[Bug]: LC0091 falsely requires translation for base language (en-US)

### ISSUE THREAD

**Issue by ralf-escher** [ALCops/Analyzers#296]
### Analyzer

LinterCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

LC0091

### Description

**Describe the bug**
`LC0091 (Translatable text should be translated)` reports missing translations for `en-US` when no explicit `LanguagesToTranslate` is defined in `alcops.json`.

In AL projects, `en-US` is the source language and already represented by generated `*.g.xlf`. Requiring an additional translated target for the base language is redundant and creates false positives.

**To Reproduce**
1. Create an AL project with:
   - `supportedLocales`: `["de-DE", "en-US"]`
   - feature `"TranslationFile"`
2. Maintain German translations in `*.de-DE.xlf`.
3. Do not create `alcops.json` with `LanguagesToTranslate`.
4. Run ALCops `0.8.3`.

**Current Result**
Diagnostics like:
- `warning LC0091: Missing translation for 'MissingFieldValueErr' in language(s): en-US`
- `warning LC0091: Missing translation for 'MissingExpectedValueErr' in language(s): en-US`
- `warning LC0091: Missing translation for 'Caption' in language(s): en-US`

**Expected Result**
LC0091 should not require base-language (`en-US`) translations by default.

`en-US` should be treated as source language and excluded from "missing translation" checks unless explicitly requested.

**Workaround**
Adding this suppresses the false positive:
`{"LanguagesToTranslate": ["de-DE"]}`

**Why this is a problem**
- Contradicts AL translation model (English source language).
- Forces extra analyzer configuration for a standard setup.
- Adds CI noise and unnecessary maintenance overhead.

**Environment**
- Business Central: `28` (`application`: `28.0.0.0`)
- ALCops: `0.8.3`
- Setup: App + Test projects, `TranslationFile` enabled, `de-DE` maintained, `*.g.xlf` generated

**Suggested fix**
1. Determine source/base language from XLF (`source-language`) or project metadata.
2. Exclude source/base language from LC0091 missing-translation diagnostics by default.
3. Keep override behavior when `LanguagesToTranslate` is explicitly configured.

### AL code to reproduce

```al

```

**Comment by Arthurvdv** [comment]
**Clarification needed on the project setup**

Thanks for reporting this. We've investigated the diagnostic and want to confirm the root cause before changing the implementation.

**How LC0091 currently discovers languages:**

The analyzer calls the SDK's `LanguageFileUtilities.GetXliffLanguageFiles()`, which returns all `*.xlf` and `*.xliff` files in the `Translations/` folder **except** the generated source file (`<AppName>.g.xlf`). It then reads the `target-language` attribute from each file to build the list of languages that need translations.

This means if a file like `<AppName>.en-US.xlf` exists in the `Translations/` folder, the analyzer treats `en-US` as a language that requires translations, even though the source text is already in English.

**Our question:**

Does your project have a `<AppName>.en-US.xlf` file in the `Translations/` folder? We suspect this is generated because `en-US` is listed in `supportedLocales` in `app.json`, which causes the AL compiler to create the target file.

If so, the challenge we face is: how should the analyzer distinguish between:
1. "This `en-US.xlf` was auto-generated by `supportedLocales` but we don't need custom English translations" (the false positive scenario you described)
2. "We intentionally have an `en-US.xlf` because we want to override source texts with custom English translations"

We're not sure how we should handle this scenario as we can't distinguish between both cases. Any input on how you'd expect this to behave would be helpful.

In the meantime, the workaround of explicitly setting `"LanguagesToTranslate": ["de-DE"]` in `alcops.json` remains the recommended approach to avoid the false positive.

---

*This comment was drafted with help of GitHub Copilot (AI)*

**Comment by ralf-escher** [comment]
@Arthurvdv no, we only use the standard `g.xlf` and `de-DE.xlf` translations.
This `LC0091` warning is quite new, therefore I use `alcops.json` meanwhile.

**Comment by Arthurvdv** [comment]
Hmm, that’s strange. I’m currently having a hard time reproducing this issue on my side.

To investigate it further, I created a small sample project (see attach [Demo.zip](https://github.com/user-attachments/files/28047621/Demo.zip)), but so far I haven’t been able to trigger the false positive. Am I perhaps missing a specific step or setup detail?

If you can share a bit more context or maybe a minimal repro, I’ll gladly take another look

**Comment by Arthurvdv** [comment]
@ralf-escher did you had a chance to look af the sample project I've created?

**Comment by Arthurvdv** [comment]
Closing this issue due to inactivity. If this is still relevant, feel free to reopen it and we can pick up the discussion again.
